firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/Ranges.txt

```diff
@@ -0,0 +1,31 @@
+Scenario: The Util Ranges  within function works correctly
+  GIVEN("The Util within function")
+  WHEN("floor is '\x01' and ceiling is '\0ff' , value is within range")
+     THEN("Returns true , as value is in range")
+
+
+Scenario:The Util Ranges  clamp function works correctly
+  GIVEN("The Util clamp function'")
+   WHEN("Value is equal to ceiling")
+      THEN("Returns correct value '\x83'")
+
+   WHEN("Value is greater than ceiling")
+     THEN("Returns expected value ceiling ")
+  
+  GIVEN("The Util clamp function ")
+  WHEN("Value is less than floor")
+    THEN("Returns expected floor value 0x0a")
+
+  WHEN("Value is equal to floor")
+    THEN("Returns expected  input value 0x08 ")  
+
+  GIVEN("The Util clamp function")  
+  WHEN("Value is in the range")
+    THEN("Returns expected  input value 0x53 ")
+
+
+  WHEN("floor is equal to ceiling")
+     THEN("Returns expected value ") 
+
+  WHEN("floor is greater than ceiling") 
```

Comment:
All of these `WHEN`s should fully specify the values of the input parameters, e.g. `WHEN('the function is called with a floor of 0x00, a ceiling of 0x83, and a pre-clamped value of 0x83')`, or `WHEN('the function is called with a floor of 0xff, a ceiling of 0x83, and a pre-clamped value of 0x83')`. The `THEN`s should specify the return value, e.g. `THEN('the function returns 0x83 as the clamped value')`